compiler/hash-typecheck/src/traverse/visitor.rs

```diff
@@ -1662,6 +1662,35 @@ impl<'tc> visitor::AstVisitor for TcVisitor<'tc> {
     ) -> Result<Self::BinaryExprRet, Self::Error> {
         let walk::BinaryExpr { lhs, rhs, .. } = walk::walk_binary_expr(self, node)?;
 
+        // @@Hack: currently, trait resolution logic is broken, so we introduce
+        // a workaround for primitives to support all operators so that we can
+        // continue onto lowering...
+        //
+        // @@Note: this does not handle floats.
+        if self.oracle().term_is_primitive(lhs) && self.oracle().term_is_primitive(rhs) {
+            let lhs_term = if self.oracle().term_is_literal(lhs) {
+                self.typer().infer_ty_of_term(lhs)?
+            } else {
+                lhs
```

Comment:
you can't just return `lhs` otherwise, because in one branch you're returning a type and in another you're returning a value

Reply:
Yes, I see. I guess it should check if it is a `value` overall and then try and get the type of it. Is this safe to do in all cases where the term is `Level0`?